src/Auth0.OidcClient.MAUI.Platforms.Windows/WebAuthenticator.cs

```diff
@@ -0,0 +1,99 @@
+
+                    if (ctx is not null && ctx.TaskId is not null)
+                    {
+                        _tasksManager.ResumeTask(protocolArgs.Uri, ctx.TaskId);
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// Begin an authentication flow by navigating to the specified url and waiting for a callback/redirect to the callbackUrl scheme.
+        /// </summary>
+        /// <param name="authorizeUri">Url to navigate to, beginning the authentication flow.</param>
+        /// <param name="callbackUri">Expected callback url that the navigation flow will eventually redirect to.</param>
+        /// <param name="cancellationToken">Cancellation token.</param>
+        /// <returns>Returns a result parsed out from the callback url.</returns>
+        /// <remarks>Prior to calling this, a call to <see cref="Activator.CheckRedirectionActivation()"/> must be made during application startup.</remarks>
+        /// <seealso cref="Activator.CheckRedirectionActivation()"/>
+        public async Task<WebAuthenticatorResult> AuthenticateAsync(Uri authorizeUri, Uri callbackUri, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            if (!_activator.RedirectActivationChecked)
+            {
+                throw new InvalidOperationException("The redirection check on app activation was not detected. Please make sure a call to Activator.CheckRedirectionActivation was made during App creation.");
+            }
+            if (!_helpers.IsAppPackaged)
+            {
+                throw new InvalidOperationException("The WebAuthenticator requires a packaged app with an AppxManifest.");
+            }
+            if (!_helpers.IsUriProtocolDeclared(callbackUri.Scheme))
+            {
+                throw new InvalidOperationException($"The URI Scheme {callbackUri.Scheme} is not declared in AppxManifest.xml.");
+            }
+
+            var redirectContext = RedirectionContext.New(_appInstanceProxy);
+
+            authorizeUri = StateModifier.WrapStateWithRedirectionContext(authorizeUri, redirectContext);
+
+            var tcs = new TaskCompletionSource<Uri>();
+
+            if (cancellationToken.CanBeCanceled)
+            {
+                // When the cancellationToken is used to trigger the cancellation
+                // we need to cancel the TaskCompletionSource and remove the taskid from the tasks.
+                cancellationToken.Register(() =>
+                {
+                    tcs.TrySetCanceled();
+                    _tasksManager.Remove(redirectContext.TaskId);
+                });
+
+                cancellationToken.ThrowIfCancellationRequested();
+            }
+
+            var newUri = authorizeUri.ToString().IndexOf("logout", StringComparison.CurrentCultureIgnoreCase) > -1 ? StateModifier.MoveStateToReturnTo(authorizeUri) : authorizeUri;
```

Comment:
We should make this check more resilient by ensuring we do not just check for the logout string, but actualy check if the full path is `v2/logout`.